modules/host/network.go

```diff
@@ -18,14 +18,14 @@ var rpcSettingsDeprecated = types.Specifier{'S', 'e', 't', 't', 'i', 'n', 'g', '
 // checks if the host's hostname has changed, and makes an updated host
 // announcement if so.
 func (h *Host) threadedUpdateHostname() {
+	err := h.tg.Add()
+	if err != nil {
+		return
+	}
+	defer h.tg.Done()
+
 	for {
-		err := h.tg.Add()
-		if err != nil {
-			return
-		}
 		h.managedLearnHostname()
-		h.tg.Done()
-
```

Comment:
hmm. Originally I grouped it this was so that we would not have a long running hold on `tg`. As it happens now, the thread will hold the `tg` indefinitely, something that seemed not great. The way it currently is does not adhere to maximum clean shutdown, but it does mean that we lose a long running hold on `tg`.